### PR TITLE
test ReadMatrixFromEnviron

### PR DESCRIPTION
**Pull request by HelloGrayson**
cc @yarpc/golang

### CODE REVIEW

File: environ_test.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	"github.com/stretchr/testify/assert"
+)
+
+func TestReadMatrixFromEnviron(t *testing.T) {
+
+	clients := []string{"yarpc-go", "yarpc-node", "yarpc-browser"}
+	servers := []string{"yarpc-go", "yarpc-node"}
+	transports := []string{"http", "tchannel"}
+
+	os.Setenv("XLANG_CLIENTS", strings.Join(clients, ","))
```
Comment (HelloGrayson) [inline environ_test.go]: I might switch this to `os.Setenv("XLANG_CLIENTS", "yarpc-go,yarpc-node,yarpc-browser")` to favor readability, and just duplicate the values in the assert statement.

**Comment by blampe** [comment]
:sheep:

**Comment by HelloGrayson** [comment]
I do not know what this meenz :P

File: Makefile [hunk 2]
```diff
@@ -5,12 +5,12 @@ project := xlang
 install:
 	glide --version || go get github.com/Masterminds/glide
 	GO15VENDOREXPERIMENT=1 glide install
-	GO15VENDOREXPERIMENT=1 go build `GO15VENDOREXPERIMENT=1 glide novendor`
+	GO15VENDOREXPERIMENT=1 go build `glide novendor`
```
Comment (prashantv) [inline Makefile:8]: you should just set this at the top of the Makefile, and then all commands should inherit it.

Comment (HelloGrayson) [inline Makefile:8]: GNU Make allows for top-level env var's? I didn't realize that... or did you just mean something like:

```
env:
    export GO15VENDOREXPERIMENT=1

install: env
    ...
```

Comment (prashantv) [inline Makefile:8]: You can do top level exports,
https://github.com/uber/tchannel-go/blob/dev/Makefile#L12

Each command for a target runs in its own sub-shell so having an `env` target won't work (it's lost after that shell). The export command is what you want to use.

Then you don't need to keep setting the variable everywhere.

**Pull request merged**